crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -98,7 +95,7 @@ impl Watcher {
 }
 
 fn is_temporary_file(path: &Path) -> bool {
-    const IGNORED_EXTENSIONS: &[&str] = &["tmp", "swp", "swo"];
+    const IGNORED_EXTENSIONS: &[&str] = &["tmp", "swp", "swo", "swx"];
```

Comment:
What type of file is this?

Reply:
my vim editor creates it ```
$ vim --version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled May 03 2024 02:37:51)
Included patches: 1-579, 1969, 580-1848, 4975, 5016, 5023, 5072, 2068, 1849-1854, 1857, 1855-1857, 1331, 1858, 1858-1859, 1873, 1860-1969, 1992, 1970-1992, 2010, 1993-2068, 2106, 2069-2106, 2108, 2107-2109, 2109-3995, 4563, 4646, 4774, 4895, 4899, 4901, 4919, 213, 1840, 1846-1847, 2110-2112, 2121
```

Reply:
Can we `sw*`?

Reply:
I'm a little bothered by the fact that we need to explicitly ignore all these temp files. Is there a way to use a system default or user-level setting? Maybe we could use the same settings as [ripgrep's automatic filtering](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#automatic-filtering)?

Reply:
It looks like the `ignore` crate might have some default filters. (ignore is what ripgrep uses)

Reply:
Looks like `ripgrep` might use a default extension list to search from rather than excluding stuff: https://github.com/BurntSushi/ripgrep/blob/ea99421ec896fcc9a9cd5367d42a80f89316822f/crates/ignore/src/default_types.rs#L274

Reply:
I tried using the `ignore` crate's `Gitignore` which is nice because it will default use .gitignore patterns to ignore which we could use to not trigger rebuilds. But you still need to add in stuff like .swp to get it to ignore those, I'm not sure if there is anything built in to ignore those files. 
```
    pub fn new(env_toml_dir: &Path, packages: &[PathBuf]) -> Self {
        let env_toml_dir: Arc<PathBuf> = Arc::new(canonicalize_path(env_toml_dir));
        let packages: Arc<Vec<PathBuf>> = Arc::new(packages.iter().map(|p| canonicalize_path(p)).collect());
        
        let mut builder = GitignoreBuilder::new(&*env_toml_dir);
        for package in packages.iter() {
            builder.add(package);
        }
        let ignores = Arc::new(builder.build().expect("Failed to build GitIgnore"));

        Self {
            env_toml_dir,
            packages,
            ignores,
        }
    }
```

Reply:
I pushed some code that utilizes any gitignore patterns and I also added the 4913 one. It will still show multiple events for something like a vim save because it is doing multiple operations in the background (including creating a 4913 file which it copies over) but the debouncing takes care of it so that it doesnt rebuild. We could also implement some debouncing on the event handler so that it only prints out the last file modified that it detected which would clean up the console output.

Reply:
That sounds great!